Refresh.GameServer/Services/SmtpService.cs

```diff
@@ -20,33 +20,36 @@ public class SmtpService : EndpointService
     {
         this._integrationConfig = integrationConfig;
         this._gameConfig = gameConfig;
-
-        if (!this._integrationConfig.SmtpEnabled) return;
-        
-        this._smtpClient = new SmtpClient(this._integrationConfig.SmtpHost)
-        {
-            Port = this._integrationConfig.SmtpPort,
-            EnableSsl = this._integrationConfig.SmtpTlsEnabled,
-            Credentials = new NetworkCredential(this._integrationConfig.SmtpUsername, this._integrationConfig.SmtpPassword),
-            DeliveryMethod = SmtpDeliveryMethod.Network,
-        };
     }
 
     private bool SendEmail(string recipient, string subject, string body)
     {
-        if (!this._integrationConfig.SmtpEnabled || this._smtpClient == null) return false;
+        if (!this._integrationConfig.SmtpEnabled) return false;
+
+        using SmtpClient client = new();
+
+        SecureSocketOptions tlsOptions = SecureSocketOptions.None;
+
+        if (this._integrationConfig is { SmtpTlsEnabled: true, SmtpPort: 587 })
+            tlsOptions = SecureSocketOptions.StartTls;
+        else if (this._integrationConfig.SmtpTlsEnabled)
+            tlsOptions = SecureSocketOptions.SslOnConnect;
```

Comment:
Is there a reason that we hardcode only 587 for TLS?
This behaviour was not in the old code

Reply:
587 is plaintext only and requires the `STARTTLS` extension, but other ports are TLS \*always\*.

The old client probably had similar logic internally.